napari/utils/events/read_only_model.py

```diff
@@ -0,0 +1,19 @@
+from pydantic import BaseModel
+
+
+class ReadOnlyModel(BaseModel):
```

Comment:
Is a Pydantic BaseModel really necessary (what is the overhead?), or do you want to use [pydantic dataclass](https://pydantic-docs.helpmanual.io/usage/dataclasses/) even maybe stand dataclasses with `frozen=True`?
Might not have validation, but it's likely a bit more lightweight.